src/libraries/Microsoft.PowerFx.Core/Public/Values/DateValue.cs

```diff
@@ -14,6 +14,28 @@ namespace Microsoft.PowerFx.Types
     /// </summary>
     public class DateValue : PrimitiveValue<DateTime>
     {
+        /// <summary>
+        /// Converts To UTC time if value is not utc and <paramref name="timeZoneInfo"/> is utc.
+        /// Converts from UTC time if  value is utc and <paramref name="timeZoneInfo"/> is non utc.
+        /// else returns the  value/>/>.
+        /// NOTE: if <paramref name="timeZoneInfo"/> is null, Local time zone is used.
+        /// </summary>
+        public DateTime GetConvertedDateValue(TimeZoneInfo timeZoneInfo)
```

Comment:
Can we share this with DateTimeValue class?
(eg, a common static helper)

Reply:
done.